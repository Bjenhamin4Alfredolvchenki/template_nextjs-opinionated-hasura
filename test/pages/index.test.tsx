/* eslint-disable @typescript-eslint/no-extra-semi */
import React from 'react'
import * as TestingLib from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Page from '../../src/pages'

global.fetch = require('node-fetch')

describe('Home page', () => {
  it('should render home page with user signed out', () => {
    const { getByText, getByRole } = TestingLib.render(<Page />)

    const eleList = getByText('HOME')
    const image = getByRole('img')
    expect(eleList).toBeInTheDocument()
    expect(image).toHaveAttribute(
      'src',
      'http://daisyui.com/tailwind-css-component-profile-1@94w.png'
    )
  })
})
