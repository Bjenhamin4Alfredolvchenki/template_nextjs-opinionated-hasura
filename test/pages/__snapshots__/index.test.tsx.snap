// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <div
      class="p-6 bg-purple-800"
    >
      <div
        class="flex flex-col py-4 font-sans bg-white"
      >
        <div
          class="container md:px-20 md:mx-auto"
        >
          <header
            class="relative flex flex-col items-center justify-between py-2 sm:flex-row"
          >
            <nav
              class="flex flex-wrap text-lg"
            >
              <a
                class="px-6 py-3 text-gray-00 hover:text-purple-300"
                href="/"
              >
                Home
              </a>
              <a
                class="px-6 py-3 text-gray-00 hover:text-purple-300"
                href="/messages"
              >
                Messages (hasura)
              </a>
              <a
                class="px-6 py-3 text-gray-00 hover:text-purple-300"
                href="https://github.com/saitodisse/nextjs-opinionated-hasura"
              >
                Github
              </a>
            </nav>
            <button
              class="absolute top-0 right-0 flex flex-col p-4 mt-5 md:hidden"
            >
              <span
                class="w-5 h-px mb-1 bg-orange-500"
              />
              <span
                class="w-5 h-px mb-1 bg-orange-500"
              />
              <span
                class="w-5 h-px mb-1 bg-orange-500"
              />
            </button>
          </header>
          <main
            class="flex flex-col justify-between pt-8 mx-8 sm:flex-row"
          >
            <div
              class="flex flex-col items-center text-center sm:w-2/5 sm:items-start sm:text-left"
            >
              <h1
                class="flex flex-col mb-2 text-6xl font-bold leading-none text-purple-900 sm:flex-row md:tracking-wide"
              >
                <svg
                  class="mr-2 text-purple-500"
                  fill="none"
                  height="60"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="60"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polygon
                    points="13 2 3 14 12 14 11 22 21 10 12 10 13 2"
                  />
                </svg>
                Next.js
              </h1>
              <h2
                class="mb-6 ml-6 text-2xl text-orange-500 sm:text-4xl sm:tracking-widest text-secondary"
              >
                opinionated
              </h2>
              <p
                class="mb-12 leading-relaxed text-gray-600"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing. Vestibulum rutrum metus at enim congue scelerisque. Sed suscipit metu non iaculis semper consectetur adipiscing elit.
              </p>
            </div>
            <div
              class="flex flex-col items-end my-3 text-right md:ml-20"
            >
              <div
                class="text-3xl font-bold text-purple-900"
              >
                Libs
              </div>
              <a
                class="my-0.5 text-blue-700 underline"
                href="https://www.typescriptlang.org/"
                rel="noreferrer"
                target="_blank"
              >
                typescript
              </a>
              <a
                class="my-0.5 text-blue-700 underline"
                href="https://nextjs.org/"
                rel="noreferrer"
                target="_blank"
              >
                next.js
              </a>
              <a
                class="my-0.5 text-blue-700 underline"
                href="https://tailwindcss.com/"
                rel="noreferrer"
                target="_blank"
              >
                tailwindcss
              </a>
              <a
                class="my-0.5 text-blue-700 underline"
                href="https://storybook.js.org/"
                rel="noreferrer"
                target="_blank"
              >
                storybook
              </a>
              <a
                class="my-0.5 text-blue-700 underline"
                href="https://jestjs.io/"
                rel="noreferrer"
                target="_blank"
              >
                jest
              </a>
              <a
                class="my-0.5 text-blue-700 underline"
                href="https://eslint.org/"
                rel="noreferrer"
                target="_blank"
              >
                eslint
              </a>
              <a
                class="my-0.5 text-blue-700 underline"
                href="https://prettier.io/"
                rel="noreferrer"
                target="_blank"
              >
                prettier
              </a>
              <a
                class="my-0.5 text-blue-700 underline"
                href="https://sweetalert2.github.io/"
                rel="noreferrer"
                target="_blank"
              >
                sweetalert2
              </a>
              <a
                class="my-0.5 text-blue-700 underline"
                href="https://react-icons.github.io/react-icons/"
                rel="noreferrer"
                target="_blank"
              >
                react-icons
              </a>
            </div>
          </main>
          <div
            class="flex flex-wrap justify-center mx-2 my-4"
          >
            <a
              class="px-6 py-3 mb-1 mr-1 underline"
              href="https://github.com/saitodisse/nextjs-opinionated-hasura"
              rel="noreferrer"
              target="_blank"
            >
              Github
            </a>
            <button
              class="px-6 py-3 mx-3 mb-1 mr-1 text-sm font-bold text-purple-700 transition-all duration-150 ease-linear bg-transparent border border-purple-500 border-solid rounded outline-none hover:bg-purple-500 hover:text-white active:bg-purple-600"
            >
              Show Image
            </button>
            <button
              class="px-6 py-3 mx-3 mb-1 mr-1 text-sm font-bold text-purple-700 transition-all duration-150 ease-linear bg-transparent border border-purple-500 border-solid rounded outline-none hover:bg-purple-500 hover:text-white active:bg-purple-600"
            >
              <span>
                Call API
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
